lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -4568,7 +4626,9 @@ void ByteCodeGenerator::EmitPropStore(Js::RegSlot rhsLocation, Symbol *sym, Iden
         }
         // Make sure the property has a slot. This will bump up the size of the slot array if necessary.
         Js::PropertyId slot = sym->EnsureScopeSlot(funcInfo);
-        bool chkBlockVar = !isLetDecl && !isConstDecl && NeedCheckBlockVar(sym, scope, funcInfo);
+        bool chkBlockVar = !isLetDecl && !isConstDecl && NeedCheckBlockVar(sym, scope, funcInfo)
+                            // Skip the check for parameters when not merged with body scope
+                            && (sym->GetScope()->GetScopeType() != ScopeType_Parameter || sym->GetScope()->GetCanMergeWithBodyScope());
```

Comment:
Can we move this check into NeedCheckBlockVar? Usually we don't need to touch EmitPropStore much, so I'm wondering if this also needs to be reflected elsewhere NeedCheckBlockVar is used.